src/Razor/test/Microsoft.AspNetCore.Razor.LanguageServer.Test/Semantic/DefaultRazorSemanticTokenInfoServiceTest.cs

```diff
@@ -6,314 +6,445 @@
 using System.Threading;
 using System.Threading.Tasks;
 using Microsoft.AspNetCore.Razor.LanguageServer.Common;
-using Microsoft.AspNetCore.Razor.LanguageServer.Extensions;
 using Microsoft.AspNetCore.Razor.LanguageServer.Formatting;
 using Microsoft.AspNetCore.Razor.LanguageServer.JsonRpc;
 using Microsoft.AspNetCore.Razor.LanguageServer.ProjectSystem;
 using Microsoft.AspNetCore.Razor.LanguageServer.Semantic;
-using Microsoft.AspNetCore.Razor.LanguageServer.Semantic.Models;
+using Microsoft.AspNetCore.Razor.Test.Common;
 using Microsoft.CodeAnalysis.Razor.ProjectSystem;
-using Microsoft.CodeAnalysis.Razor.Workspaces.Extensions;
 using Moq;
 using OmniSharp.Extensions.JsonRpc;
-using OmniSharp.Extensions.LanguageServer.Protocol.Models;
 using Xunit;
-using OmniSharpRange = OmniSharp.Extensions.LanguageServer.Protocol.Models.Range;
+using OS = OmniSharp.Extensions.LanguageServer.Protocol.Models;
 
 namespace Microsoft.AspNetCore.Razor.LanguageServer.Test.Semantic
 {
     // Sets the FileName static variable.
     // Finds the test method name using reflection, and uses
     // that to find the expected input/output test files as Embedded resources.
     [IntializeTestFile]
+    [UseExportProvider]
     public class DefaultRazorSemanticTokenInfoServiceTest : SemanticTokenTestBase
     {
         #region CSharp
         [Fact]
         public async Task GetSemanticTokens_CSharp_RazorIfNotReady()
         {
-            var txt = $@"<p></p>@{{
+            var documentText = @"<p></p>@{
     var d = ""t"";
-}}";
+}
+";
+
+            var razorRange = new OS.Range
+            {
+                Start = new OS.Position { Line = 0, Character = 0 },
+                End = new OS.Position { Line = 3, Character = 0 }
+            };
 
-            var cSharpResponse = new ProvideSemanticTokensResponse(
-                tokens: Array.Empty<int>(), isFinalized: true, hostDocumentSyncVersion: 1);
-            await AssertSemanticTokensAsync(txt, isRazor: false, csharpTokens: cSharpResponse, documentVersion: 1);
+            var csharpTokens = new ProvideSemanticTokensResponse(
+                tokens: Array.Empty<int>(), isFinalized: false, hostDocumentSyncVersion: 1);
+            await AssertSemanticTokensAsync(documentText, isRazorFile: false, razorRange, csharpTokens: csharpTokens, documentVersion: 1);
         }
 
         [Fact]
         public async Task GetSemanticTokens_CSharpBlock_HTML()
         {
-            var txt = @$"@{{
+            var documentText = @"@{
     var d = ""t"";
     <p>HTML @d</p>
-}}";
+}
+";
 
-            var csharpTokens = new int[]
+            var razorRange = new OS.Range
             {
-                20, 4, 3, RazorSemanticTokensLegend.CSharpKeyword, 0, // var
```

Comment:
🥂 🎉 😍

Reply:
Lol yeah, pretty nice to not have to keep all these manually up to date.